ledger/participant-integration-api/src/main/scala/platform/apiserver/services/tracking/TrackerMap.scala

```diff
@@ -61,16 +61,36 @@ private[services] final class TrackerMap[Key](
       executionContext: ExecutionContext,
       loggingContext: LoggingContext,
   ): Future[Either[TrackedCompletionFailure, CompletionSuccess]] = {
+
+    def cleanClosed(
+        submitter: Key,
+        trackerResource: AsyncResource[TrackerWithLastSubmission],
+    ): Option[AsyncResource[TrackerWithLastSubmission]] = {
+      trackerResource.currentState match {
+        case Ready(tracker) if tracker.isCompleted =>
```

Comment:
Can this happen? (that a tracker is still ready, but already done?) If yes shouldn't we do the same in the cleanup?

Reply:
Yes, that's what happens when the akka is torn down. It is torn down when it fails to recover after upstream failure. There is no-one else to change the overlapping resource status so we end up in this odd Ready/isCompleted state.
And yes you are right, I should be cleaning up also in the cleanup :+1:

Reply:
great, thanks!